sqlx-sqlite/src/connection/mod.rs

```diff
@@ -557,6 +611,64 @@ impl LockedSqliteHandle<'_> {
         let ret = unsafe { sqlite3_get_autocommit(self.as_raw_handle().as_ptr()) };
         ret == 0
     }
+
+    /// Sets the soft heap limit for the current thread.
+    ///
+    /// This function sets a soft limit on the amount of heap memory that can be allocated by SQLite.
+    /// The limit is in bytes. If `limit` is zero, the heap limit is disabled.
+    ///
+    /// Returns the previous heap limit. If the heap limit was never set, returns 0.
+    ///
+    /// See: https://www.sqlite.org/c3ref/hard_heap_limit64.html
+    pub fn soft_heap_limit(&mut self, limit: i64) -> i64 {
+        unsafe { sqlite3_soft_heap_limit64(limit) }
```

Comment:
The very first line of the linked documentation says:

>  These interfaces impose limits on the amount of heap memory that will be by all database connections within a single process.

So saying it sets it for the current thread, or having it as a method on the connection. is very misleading.

It should be a free function, or perhaps an associated method of the `Sqlite` type. Probably a free function.

Passing a negative number just gets the heap limit and doesn't set it, which also needs to be documented.

A more Rust-y API would be something like `limit: Option<u64>`, returning an error if it's greater than `i64::MAX`, and providing a separate getter function, but I don't know if we need to go that far. I'd kind of like to sleep on it, though.